src/metabase/models/dashboard.clj

```diff
@@ -18,17 +19,53 @@
   (db/cascade-delete! 'Revision :model "Dashboard" :model_id id)
   (db/cascade-delete! DashboardCard :dashboard_id id))
 
+(defn- pre-insert [dashboard]
+  (let [defaults {:parameters []}]
+    (merge defaults dashboard)))
 
 (i/defentity Dashboard :report_dashboard)
 
 (u/strict-extend (class Dashboard)
   i/IEntity
   (merge i/IEntityDefaults
          {:timestamped?       (constantly true)
-          :types              (constantly {:description :clob})
+          :types              (constantly {:description :clob, :parameters :json})
           :can-read?          i/publicly-readable?
           :can-write?         i/publicly-writeable?
-          :pre-cascade-delete pre-cascade-delete}))
+          :pre-cascade-delete pre-cascade-delete
+          :pre-insert         pre-insert}))
+
+
+;;; ## ---------------------------------------- PERSISTENCE FUNCTIONS ----------------------------------------
+
+
+(defn create-dashboard
+  "Create a `Dashboard`"
+  [{:keys [name description parameters public_perms], :as dashboard} user-id]
+  {:pre [(map? dashboard)
+         (u/nil-or-maplist? parameters)
+         (integer? user-id)]}
+  (->> (db/insert! Dashboard
+                   :name         name
+                   :description  description
+                   :parameters   (or parameters [])
+                   :public_perms public_perms
+                   :creator_id   user-id)
+       (events/publish-event :dashboard-create)))
+
+(defn update-dashboard
+  "Update a `Dashboard`"
```

Comment:
These docstrings need some work. What's the purpose of this function? Why should I use it vs. just calling `db/update!`? Can I still use `db/update!` to update a `Dashboard`, or should I always use `update-dashboard`? What does `update-dashboard` do differently from `db/update!`?

Also don't forget functions that have side effects like these need to end in a `!`.